packages/trail-fastify-graphql-plugin/lib/graphql.js

```diff
@@ -0,0 +1,208 @@
+      const value = GraphQLJSON.parseLiteral(ast)
+      if (typeof value === 'object' && typeof value.id === 'string') {
+        return value
+      }
+    }
+    return null
+  }
+})
+
+const SortOrder = {
+  ID_ASC: 'id',
+  WHEN_ASC: 'when',
+  WHO_ASC: 'who',
+  WHAT_ASC: 'what',
+  SUBJECT_ASC: 'subject',
+  ID_DESC: '-id',
+  WHEN_DESC: '-when',
+  WHO_DESC: '-who',
+  WHAT_DESC: '-what',
+  SUBJECT_DESC: '-subject'
+}
+
+const TrailType = {
+  WHO: 'who',
+  WHAT: 'what',
+  SUBJECT: 'subject'
+}
+
+const typeDefs = `
+  scalar Date
+  scalar StringWithAttrs
+  scalar JSON
+
+  enum SortOrder { ${Object.keys(SortOrder).join(' ')} }
+
+  enum TrailType { ${Object.keys(TrailType).join(' ')} }
+
+  type Trail {
+    id: Int!
+    when: Date!
+    who: StringWithAttrs!
+    what: StringWithAttrs!
+    subject: StringWithAttrs!
+    where: JSON
+    why: JSON
+    meta: JSON
+  }
+
+  type Query {
+
+    trail(id: Int!): Trail
+
+    trails(
+      from: Date!
+      to: Date!
+      who: String
+      what: String
+      subject: String
+      page: Int
+      pageSize: Int
+      sort: SortOrder
+      exactMatch: Boolean
+      caseInsensitive: Boolean
+    ): [Trail!]!
+
+    enumerate(
+      from: Date!
+      to: Date!
+      type: TrailType!
+      page: Int
+      pageSize: Int
+      descending: Boolean
+    ): [String!]!
+
+  }
+
+  type Mutation {
+
+    insert(
+      when: Date!
+      who: StringWithAttrs!
+      what: StringWithAttrs!
+      subject: StringWithAttrs!
+      where: JSON
+      why: JSON
+      meta: JSON
+    ): Trail
+
+    update(
+      id: Int!
+      when: Date
+      who: StringWithAttrs
+      what: StringWithAttrs
+      subject: StringWithAttrs
+      where: JSON
+      why: JSON
+      meta: JSON
+    ): Boolean!
+
+    remove(id: Int!): Boolean!
+
+  }
+`
+
+function makeResolvers (opts) {
+  const {
+    logger,
+    pool,
+    trailsManager = new TrailsManager(logger, pool)
+  } = opts
+
+  return {
+    Query: {
+      trail (_, { id }) {
+        return trailsManager.get(id)
+      },
+      trails (_, args) {
+        return trailsManager.search(args)
+      },
+      enumerate (_, args) {
+        return trailsManager.enumerate(args)
+      }
+    },
+    Mutation: {
+      async insert (_, trail) {
+        const id = await trailsManager.insert(trail)
+        return id ? trailsManager.get(id) : null
+      },
+      async update (_, { id, ...trail }) {
+        const record = await trailsManager.get(id)
+        if (!record) {
+          return false
+        }
+        const { id: x, ...fields } = record
+        return trailsManager.update(id, { ...fields, ...trail })
+      },
+      // NOTE: This resolver should be called 'delete' but graphql-jit has problems
+      // with the name (presumably because of clash with the JS 'delete' keyword).
+      remove (_, { id }) {
+        return trailsManager.delete(id)
+      }
+    },
+    Date,
+    StringWithAttrs,
+    JSON: GraphQLJSON,
```

Comment:
This means that it will be a string with JSON content?

Reply:
No, it means you can do something like this in a query:
```
insert( ... meta: { tags: [ 1, "xxx" ] })
```
And the value will be serialized to JSON in the trails DB and deserialized back to an object value in a query result.

I've just added a unit test demonstrating this usage, and whilst the test works as expected, graphql-jit prints the following error when it encounters a query argument in that format:
```
Scalar with variable inputs detected for parsing AST literals. This is not supported.
```
It doesn't seem to impact query functionality, but maybe it would be better to avoid the error by switching to JSON in string values?

Reply:
Looking at the graphql-jit source code, it prints the error message when it encounters a particular condition but then proceeds as normal anyway, so the error message is really just a warning but has nuisance value.

Reply:
Can you open an issue in graphql-jit for this? Let's disable the jit for now.

Reply:
Found a work around for the problem.